websites/I/ITVX/presence.ts

```diff
@@ -0,0 +1,188 @@
+interface ChannelSlotData {
+	displayTitle: string;
+	start: string;
+	end: string;
+	seriesNumber?: number;
+	episodeNumber?: number;
+}
+
+interface ChannelMetadata {
+	id: string;
+	name: string;
+	slots: {
+		now: ChannelSlotData;
+		next: ChannelSlotData;
+	};
+	slug: string;
+}
+
+interface LiveChannelNextData {
+	props: {
+		pageProps: {
+			channelSlug: string;
+			channelsMetaData: {
+				channels: ChannelMetadata[];
+			};
+		};
+	};
+}
+
+interface CategoriesNextData {
+	props: {
+		pageProps: {
+			category: {
+				slug: string;
+				title: string;
+			};
+		};
+	};
+}
+
+interface ProgrammeNextData {
+	props: {
+		pageProps: {
+			title: {
+				CTAText?: string;
+				episodeNumber?: number;
+				programmeTitle: string;
+				seriesNumber?: number;
+				titleType: string;
+			};
+		};
+	};
+}
+
+const presence = new Presence({
+		clientId: "1043125926793318420",
+	}),
+	fetchLiveChannelNextData = (): LiveChannelNextData => {
+		const nextDataElement = document.querySelector("#__NEXT_DATA__");
+
+		if (!nextDataElement) return null;
+
+		return JSON.parse(nextDataElement.textContent);
+	},
+	fetchCategoriesNextData = (): CategoriesNextData => {
+		const nextDataElement = document.querySelector("#__NEXT_DATA__");
+
+		if (!nextDataElement) return null;
+
+		return JSON.parse(nextDataElement.textContent);
+	},
+	fetchProgrammeNextData = (): ProgrammeNextData => {
+		const nextDataElement = document.querySelector("#__NEXT_DATA__");
+
+		if (!nextDataElement) return null;
+
+		return JSON.parse(nextDataElement.textContent);
+	};
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "itv_logo",
```

Comment:
Imgur URLs are preferred over Discord asset names now, replace.
```suggestion
			largeImageKey: "https://i.imgur.com/XisKvdg.png",
```